### PR TITLE
Major UI Enhancements using Material UI and Bug Fixes for Mentor-Mentee Profiles

### PR DESCRIPTION
**Pull request by Domm6**
### Description:
This update introduces significant improvements to the UI and functionality of both mentor and mentee profiles. Key changes include:

- Addition of headers to mentor and mentee profile pages.
- Consistent UI components across mentor and mentee views.
- Multiple styling fixes for an enhanced user experience.
- Resolved bugs related to the persistence of the "years of experience" and "time" fields in forms.
- Integration of Material UI modals in mentee profiles and the Match Modal.
- Updated schedule meeting modal with automatic titles for meetings without specified titles.
- Removal of unnecessary commented code for cleaner and more maintainable codebase.

These enhancements aim to provide a more cohesive and user-friendly interface for our users, ensuring a smoother interaction with the platform.

Images:
Screenshot 2024-07-17 at 10 50 02 AM
Screenshot 2024-07-17 at 10 50 29 AM
Screenshot 2024-07-17 at 10 50 37 AM
Screenshot 2024-07-17 at 10 56 25 AM
Screenshot 2024-07-17 at 10 56 33 AM

### CODE REVIEW

File: Frontend/src/components/meeting/MeetingModal.jsx [hunk 1]
```diff
@@ -14,35 +14,57 @@ function MeetingModal({
...
-              <p>No attendees</p>
+              <Typography>No attendees</Typography>
             )}
-            <div className="mm-btns">
+            <Box className="mm-btns" mt={2}>
               <Button
                 variant="contained"
                 color="primary"
                 onClick={() => acceptMeeting(selectedMeeting.id)}
+                sx={{ mr: 1 }}
```
Comment (wspalding) [inline Frontend/src/components/meeting/MeetingModal.jsx]: what are "mt" "sx" and "mr" ?

File: Frontend/src/components/meeting/MeetingModal.jsx [hunk 2]
```diff
@@ -14,35 +14,57 @@ function MeetingModal({
...
-            ×
-          </span>
-          <div className="mm-body">
-            <h3>{selectedMeeting.title}</h3>
-            <p>Status: {selectedMeeting.status}</p>
-            <p>Mentor: {selectedMeeting.mentorName}</p>
-            <p>Attendees:</p>
+      <Modal open={true} onClose={toggleModal}>
+        <Box
+          sx={{
```
Comment (wspalding) [inline Frontend/src/components/meeting/MeetingModal.jsx:25]: lets give this a better name

File: Frontend/src/components/mentee/mentee-profile/MenteeProfile.jsx [hunk 3]
```diff
@@ -185,7 +185,38 @@ function MenteeProfile() {
               </Typography>
             </div>
           </div>
-          {isModalOpen && (
+          <Modal open={isModalOpen} onClose={handleModalToggle}>
+            <Box
+              sx={{
```
Comment (wspalding) [inline Frontend/src/components/mentee/mentee-profile/MenteeProfile.jsx:190]: lets give this a better name

File: Frontend/src/components/mentor/mentor-profile/MentorProfileModal.jsx [hunk 4]
```diff
@@ -51,6 +51,8 @@ function MentorProfileModal({
   const [selectedSchool, setSelectedSchool] = useState(null);
   const [errorMessage, setErrorMessage] = useState("");
 
+  console.log(mentorData);
```
Comment (wspalding) [inline Frontend/src/components/mentor/mentor-profile/MentorProfileModal.jsx]: remove log

File: Frontend/src/components/mentor/mentor-profile/MentorProfile.jsx [hunk 5]
```diff
@@ -208,7 +208,38 @@ function MentorProfile() {
...
+                handleCheckboxChange={handleCheckboxChange}
+                skillsList={skillsList}
+                closeModal={() => {
+                  handleModalToggle();
+                  fetchMentorData();
+                }}
+              />
+            </Box>
+          </Modal>
+          {/* {isModalOpen && (
```
Comment (wspalding) [inline Frontend/src/components/mentor/mentor-profile/MentorProfile.jsx]: remove commented code (or add a todo to remove it, if theres a really good reason to keep it around for now)

**Comment by wspalding** [comment]
some naming issues and cleanup to be done, but overall looks pretty good

File: Frontend/src/App.jsx [hunk 6]
```diff
@@ -52,57 +55,62 @@ function App() {
...
-                  <Match />
-                </ProtectedRoute>
-              }
-            />
-          </Routes>
-        </Router>
-      </UserContext.Provider>
-    </div>
+    <ThemeProvider theme={theme}>
+      {" "}
```
Comment (sbrina813) [inline Frontend/src/App.jsx]: {" "} ?

File: Frontend/src/App.jsx [hunk 7]
```diff
@@ -52,57 +55,62 @@ function App() {
...
+      {/* Wrap with ThemeProvider */}
+      <CssBaseline /> {/* Apply CssBaseline for consistent styling */}
```
Comment (sbrina813) [inline Frontend/src/App.jsx]: can also remove these comments.

File: Frontend/src/components/calendar/CalendarModal.jsx [hunk 8]
```diff
@@ -337,72 +349,89 @@ function CalendarModal({ toggleModal, onMeetingScheduled, isMentor }) {
...
   return (
-    <div className="modal">
-      <div className="modal-content">
-        <span className="modal-close" onClick={toggleModal}>
-          ×
-        </span>
-        <h3>Schedule Meeting:</h3>
+    <Modal open={true} onClose={toggleModal}>
+      <Box
+        sx={{
```
Comment (sbrina813) [inline Frontend/src/components/calendar/CalendarModal.jsx]: Feel mix inline style with css file is not a good practice

Comment (Domm6) [reply Frontend/src/components/calendar/CalendarModal.jsx]: Resolved

File: Frontend/src/components/meeting/MeetingModal.jsx (hunk 2 above)
Comment (Domm6) [reply Frontend/src/components/meeting/MeetingModal.jsx:25]: What exactly should I name better? The modal, box, or something else?

**Pull request merged**